sympy/solvers/solveset.py

```diff
@@ -978,6 +985,365 @@ def _solveset(f, symbol, domain, _check=False):
     return result
 
 
+def _expo_solver(f):
+    r"""
+    Helper function for solving (supported) exponential equations.
+
+    Exponential equations are a sum of (currently) at most
+    2 terms with one or both of them having a power with a
+    symbol-dependent exponent.
+
+    For example
+
+    .. math:: 5^{(2*x + 3)} - 5^{(3*x - 1)}
+
+    .. math:: 4^{(5 - 9*x)} - exp^{(2 - x)}
+
+
+    The function reduces the equation (`f`) to a better log form
+    (if possible) and returns the modified equation which can be
+    further handled by `solveset`.
+
+    Examples
+    ========
+
+    >>> from sympy.solvers.solveset import _expo_solver as expo_solver
+    >>> from sympy import symbols
+    >>> x = symbols('x', real=True)
+    >>> expo_solver(3**(2*x) - 2**(x + 3))
+    2*x*log(3) - (x + 3)*log(2)
+
+
+    * Proof of correctness of the method
+
+    The logarithm is the inverse function to exponentiation.
+    The defining relation between exponentiation and logarithm is:
+
+    .. math:: {log_b x} = y \enspace if \enspace b^y = x
+
+    Therefore if we are given an equation with exponent terms, we can
+    convert every term to its corresponding log form. This is achieved by
+    taking logarithms and expanding it using log identities so that the
+    equation can be easily handled by `solveset`.
+
+    For example:
+
+    .. math:: 3^{2*x} = 2^{(x + 3)}
+
+    Taking log both sides will reduce the equation to
+
+    .. math:: (2*x)*log(3) = (x + 3)*log(2)
+
+    This form can be easily handed by `solveset`.
+    """
+
+    try:
```

Comment:
Could you make the assertion more direct? It is not transparently obvioius that `ordered` won't raise a ValueError for some reason. How about:

```
if not (isinstance(f, Add) and len(f.args) == 2:
    # solving for the sum of more than two powers is possible
    # but not yet implemented
    return None
```

Reply:
Yeah I thought too about this, but assumed that we have stated in the documentation so it would be obvious. I guess we can do it as you said.

Reply:
trust (that the user read documentation) and verify (that they obeyed the rules) :-)